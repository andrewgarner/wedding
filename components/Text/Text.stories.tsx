import { Meta, Story } from "@storybook/react";
import React from "react";
import { Text, TextProps } from "./Text";

export default {
  title: "Typography/Text",
  component: Text,
} as Meta;

const Template: Story<TextProps> = (args) => (
  <Text {...args}>The quick brown fox jumps over the lazy dog</Text>
);

export const Default = Template.bind({});
Default.args = {};
