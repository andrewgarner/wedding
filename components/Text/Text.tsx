import React, { ElementType, FunctionComponent } from "react";

enum Typography {
  HeaderXLarge = "Header x-large",
  HeaderLarge = "Header large",
  HeaderMedium = "Header medium",
  HeaderSmall = "Header small",
  Base = "Base",
  Small = "Small",
}

const typographyClassNames: Record<Typography, string> = {
  [Typography.HeaderXLarge]:
    "space-y-32 leading-loose  first-of-type:mt-0 mt-32 mb-4 mx-auto font-sans text-3xl lg:text-4xl xl:text-5xl font-normal uppercase",
  [Typography.HeaderLarge]:
    "space-y-32 leading-loose  first-of-type:mt-8 mt-32 mb-4 mx-auto font-sans text-4xl lg:text-6xl xl:text-7xl font-normal uppercase",
  [Typography.HeaderMedium]:
    "space-y-32 leading-loose  first-of-type:mt-8 mt-32 mb-4 mx-auto font-sans text-3xl lg:text-4xl font-normal uppercase",
  [Typography.HeaderSmall]:
    "space-y-32 leading-loose mt-16 mb-4 mx-auto font-sans text-2xl lg:text-3xl font-normal uppercase",
  [Typography.Base]:
    "space-y leading-loose max-w-prose mx-auto font-sans text-lg lg:text-xl font-normal",
  [Typography.Small]:
    "mx-auto max-w-prose font-sans text-sm lg:text-base font-normal",
};

interface TextProps {
  tag?: "p" | "h1" | "h2" | "h3" | "h4" | "li";
  typography?: Typography;
}

const Text: FunctionComponent<TextProps> = ({
  tag = "p",
  typography = Typography.Base,
  children,
}) => {
  const TextTag = tag as ElementType;
  return (
    // eslint-disable-next-line react/forbid-component-props
    <TextTag className={typographyClassNames[typography]}>{children}</TextTag>
  );
};

export { Text, Typography };
export type { TextProps };
