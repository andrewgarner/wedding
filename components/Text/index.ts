export { Text, Typography } from "./Text";
