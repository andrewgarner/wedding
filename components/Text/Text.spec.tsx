import { render } from "@testing-library/react";
import React from "react";

import { Text, Typography } from "./Text";

describe("The text component", () => {
  it("renders the default correctly", () => {
    const { container } = render(
      <Text>The quick brown fox jumps over the lazy dog</Text>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders the specified tag correctly", () => {
    const { container } = render(
      <Text tag="li">The quick brown fox jumps over the lazy dog</Text>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders small typography correctly", () => {
    const { container } = render(
      <Text typography={Typography.Small}>
        The quick brown fox jumps over the lazy dog
      </Text>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders x-large header typography correctly", () => {
    const { container } = render(
      <Text typography={Typography.HeaderXLarge}>
        The quick brown fox jumps over the lazy dog
      </Text>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders large header typography correctly", () => {
    const { container } = render(
      <Text typography={Typography.HeaderLarge}>
        The quick brown fox jumps over the lazy dog
      </Text>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders medium header typography correctly", () => {
    const { container } = render(
      <Text typography={Typography.HeaderMedium}>
        The quick brown fox jumps over the lazy dog
      </Text>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders small header typography correctly", () => {
    const { container } = render(
      <Text typography={Typography.HeaderSmall}>
        The quick brown fox jumps over the lazy dog
      </Text>
    );

    expect(container).toMatchSnapshot();
  });
});
