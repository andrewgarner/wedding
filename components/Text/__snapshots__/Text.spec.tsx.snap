// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The text component renders large header typography correctly 1`] = `
<div>
  <p
    class="space-y-32 leading-loose  first-of-type:mt-8 mt-32 mb-4 mx-auto font-sans text-4xl lg:text-6xl xl:text-7xl font-normal uppercase"
  >
    The quick brown fox jumps over the lazy dog
  </p>
</div>
`;

exports[`The text component renders medium header typography correctly 1`] = `
<div>
  <p
    class="space-y-32 leading-loose  first-of-type:mt-8 mt-32 mb-4 mx-auto font-sans text-3xl lg:text-4xl font-normal uppercase"
  >
    The quick brown fox jumps over the lazy dog
  </p>
</div>
`;

exports[`The text component renders small header typography correctly 1`] = `
<div>
  <p
    class="space-y-32 leading-loose mt-16 mb-4 mx-auto font-sans text-2xl lg:text-3xl font-normal uppercase"
  >
    The quick brown fox jumps over the lazy dog
  </p>
</div>
`;

exports[`The text component renders small typography correctly 1`] = `
<div>
  <p
    class="mx-auto max-w-prose font-sans text-sm lg:text-base font-normal"
  >
    The quick brown fox jumps over the lazy dog
  </p>
</div>
`;

exports[`The text component renders the default correctly 1`] = `
<div>
  <p
    class="space-y leading-loose max-w-prose mx-auto font-sans text-lg lg:text-xl font-normal"
  >
    The quick brown fox jumps over the lazy dog
  </p>
</div>
`;

exports[`The text component renders the specified tag correctly 1`] = `
<div>
  <li
    class="space-y leading-loose max-w-prose mx-auto font-sans text-lg lg:text-xl font-normal"
  >
    The quick brown fox jumps over the lazy dog
  </li>
</div>
`;

exports[`The text component renders x-large header typography correctly 1`] = `
<div>
  <p
    class="space-y-32 leading-loose  first-of-type:mt-0 mt-32 mb-4 mx-auto font-sans text-3xl lg:text-4xl xl:text-5xl font-normal uppercase"
  >
    The quick brown fox jumps over the lazy dog
  </p>
</div>
`;
